From 416ea47c992fb3217134858a52b42f315fc35778 Mon Sep 17 00:00:00 2001
From: Valentin Caron <valentin.caron@foss.st.com>
Date: Fri, 9 Jun 2023 18:25:56 +0200
Subject: [PATCH 0374/1141] dt-bindings: rtc: stm32: introduce new
 st,stm32mp25-rtc compatible

Introduce new st,stm32mp25-rtc compatible. It is based on st,stm32mp1-rtc.

Difference is that stm32mp25 soc implements a triple protection on RTC
registers:
- secure bit based protection
- privileged context based protection
- CID filtering based protection
This driver will now check theses configurations before probing to avoid
exceptions and fake reads on register.

Signed-off-by: Valentin Caron <valentin.caron@foss.st.com>
Change-Id: I73fcd6ffdf6f4aa1621547154469026bbd52ac15
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/312304
Reviewed-by: Amelie DELAUNAY <amelie.delaunay@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Amelie DELAUNAY <amelie.delaunay@foss.st.com>
---
 Documentation/devicetree/bindings/rtc/st,stm32-rtc.yaml | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/Documentation/devicetree/bindings/rtc/st,stm32-rtc.yaml b/Documentation/devicetree/bindings/rtc/st,stm32-rtc.yaml
index 71e02604e8de..d3dba2ad54cc 100644
--- a/Documentation/devicetree/bindings/rtc/st,stm32-rtc.yaml
+++ b/Documentation/devicetree/bindings/rtc/st,stm32-rtc.yaml
@@ -15,6 +15,7 @@ properties:
       - st,stm32-rtc
       - st,stm32h7-rtc
       - st,stm32mp1-rtc
+      - st,stm32mp25-rtc
 
   reg:
     maxItems: 1
@@ -116,7 +117,9 @@ allOf:
       properties:
         compatible:
           contains:
-            const: st,stm32mp1-rtc
+            anyOf:
+              - const: st,stm32mp1-rtc
+              - const: st,stm32mp25-rtc
 
     then:
       properties:
-- 
2.39.2

